packages/daemon/src/utils/wallet.ts

```diff
@@ -393,3 +399,114 @@ export const validateAddressBalances = async (mysql: MysqlConnection, addresses:
     assert.strictEqual(Number(addressBalance.unlockedBalance + addressBalance.lockedBalance), Number(addressTxHistorySum.balance));
   }
 };
+
+/**
+ * Get a list of wallet balance per token by informed transaction.
+ *
+ * @param mysql
+ * @param tx - The transaction to get related wallets and their token balances
+ * @returns
+ */
+export const getWalletBalancesForTx = async (mysql: MysqlConnection, tx: Transaction): Promise<StringMap<WalletBalanceValue>> => {
+  const addressBalanceMap: StringMap<TokenBalanceMap> = getAddressBalanceMap(tx.inputs, tx.outputs);
+  // return only wallets that were started
+  const addressWalletMap: StringMap<Wallet> = await getAddressWalletInfo(mysql, Object.keys(addressBalanceMap));
+
+  // Create a new map focused on the walletId and storing its balance variation from this tx
+  const walletsMap: StringMap<WalletBalance> = {};
+
+  // Accumulation of tokenId to be used to extract its symbols.
+  const tokenIdAccumulation = [];
+
+  // Iterates all the addresses to populate the map's data
+  const addressWalletEntries = stringMapIterator(addressWalletMap) as [string, Wallet][];
```

Comment:
We don't need to cast `as [string, Wallet][]` anymore since `stringMapIterator` uses generic typing the `addressWalletEntries` will be `[string, Wallet][]`.

<img width="416" alt="image" src="https://github.com/HathorNetwork/hathor-wallet-service-sync_daemon/assets/15909384/71657f60-60d2-4229-a06c-8b488d785fbd">

Reply:
![](https://media1.giphy.com/media/v1.Y2lkPTc5MGI3NjExNmtjZW02MmdtOWs4eTE1cHYxZjQzY2ZvdDFkemt3dWtwcTJvamo1ZCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/yJFeycRK2DB4c/giphy.gif)

Done in 08356ae607650478185b1e1d1996df5ff6e2c15c